src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1922,12 +1923,89 @@ static bool needsTupleElementNamesAttribute(TypeSymbol type)
                 }
 
                 var resultType = type.VisitType(
-                    predicate: (t, a, b) => !t.TupleElementNames.IsDefaultOrEmpty && !t.IsErrorType(),
+                    predicate: (t, _, _, _) => !t.TupleElementNames.IsDefaultOrEmpty && !t.IsErrorType(),
                     arg: (object?)null);
                 return resultType is object;
             }
         }
 
+        private void CheckUnboundedExtensionErasure(BindingDiagnosticBag diagnostics)
+        {
+            if (this.GetExtendedTypeNoUseSiteDiagnostics(null) is { } extendedType
+                && foundUnboundedErasure(extendedType, extensionsBeingErased: ConsList<TypeSymbol>.Empty.Prepend(this)))
+            {
+                diagnostics.Add(ErrorCode.ERR_CircularBase, Locations[0], extendedType, this);
+            }
+            return;
+
+            static bool foundUnboundedErasure(TypeSymbol type, ConsList<TypeSymbol> extensionsBeingErased, bool isContainer = false)
+            {
+                if (type is NamedTypeSymbol)
+                {
+                    if (!isContainer && type.IsExtension)
+                    {
+                        if (extensionsBeingErased.Contains(type.OriginalDefinition))
+                        {
+                            return true;
+                        }
+
+                        if (type.OriginalDefinition.GetExtendedTypeNoUseSiteDiagnostics(null) is not { } extendedType)
+                        {
+                            return true;
```

Comment:
>return true; [](http://example.com/codeflow?start=28&length=12)

I guess it is true that this is an error situation, but do we have to trat it as a circular dependency? Could we just ignore such type or treat it as a regular type for the purpose of the check?

Reply:
Changed to ignore the type as we discussed. That works